backend/webdav/webdav.go

```diff
@@ -482,6 +501,18 @@ func NewFs(ctx context.Context, name, root string, m configmap.Mapper) (fs.Fs, e
 			rt: ntlmssp.Negotiator{RoundTripper: t},
 		}
 	}
+	if opt.Vendor == "other" {
+		// If available, prefer Digest authentication.
+		if checkForDigestAuth(opt.URL) {
```

Comment:
This does an extra http transaction each time the backend is started which isn't ideal.

Could we fallback to digest if the transaction fails? That should give us the www authenticate header.

Or we could store the preference or digest Auth in the config and not probe for it, or maybe combine with above suggestion?

Reply:
As a matter of fact, that was my first idea/attempt. Initially, I didn't find a great place where to implement it. I'm not very familiar with rclone codebase.

However, after looking around, I'll try again in the coming days by implementing the fallback approach into `shouldRetry()`. I would appreciate any guidance you may have.

I don't have a strong opinion against adding a dedicated config param, but I think it would be nicer to automatically adapt to the server's authentication schemes. Let me give it a try. Let's see how it looks.

Reply:
Yes I agree auto adapting would be great.

shouldRetry sounds like a good place to put the logic.

Note that you can store automatically discovered stuff in the config file easily so that could be an option.